File: solutions/2947. Count Beautiful Substrings I/2947.py
# Time complexity: O(n^2)
# Space complexity: O(1)
class Solution:
    def beautifulSubstrings(self, s: str, k: int) -> int:
        ans = 0
        n = len(s)

        for i in range(n):
            vowels = 0
            consonants = 0
            for j in range(i, n):
                if self.is_vowel(s[j]):
                    vowels += 1
                else:
                    consonants += 1

                if vowels == consonants and (vowels * consonants) % k == 0:
                    ans += 1

        return ans

    def is_vowel(self, c: str) -> bool:
        return c in ['a', 'e', 'i', 'o', 'u']

File: solutions/2947. Count Beautiful Substrings I/2947.cpp
// Time complexity: O(n^2)
// Space complexity: O(1)
class Solution {
public:
    int beautifulSubstrings(string_view s, const int k) {
        int ans = 0;
        int n = s.length();

        for (int i = 0; i < n; ++i) {
            int vowels = 0, consonants = 0;
            for (int j = i; j < n; ++j) {
                if (isVowel(s[j]))
                    ++vowels;
                else 
                    ++consonants;

                if (vowels == consonants && (vowels * consonants) % k == 0)
                    ++ans;        
            }
        }

        return ans;       
    }
private:
    bool isVowel(const char &c) {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
    }     
};